### PR TITLE
Allow passing username & password as flags to broker register

### LINKED ISSUE #2479: svcat: support user/password on the cmd line
IMO svcat is about making life easier for end users, and in many cases that means allowing them to skip being exposed to the kube resource model (and yaml) for as much as possible. Registering a broker should make life easier for them by allowing them to (optionally) specify a user and password on the cmd line. svcat can then create the secret for them before it lets the rest of svc-cat do its job.

It should still support a reference to a secret as well though.

### PR DESCRIPTION
**Pull request by anmolbabu**
This PR is a 
 - [x] Feature Implementation
 - [ ] Bug Fix
 - [ ] Documentation

**What this PR does / why we need it**:

This fix adds username and password as optional parameters to broker register command.
If passed the kubernetes secret is auto created and used for broker register.

**Which issue(s) this PR fixes** 

Fixes #2479

Please leave this checklist in the PR comment so that maintainers can ensure a good PR.

Merge Checklist:
 - [ ] New feature 
   - [ ] Tests
   - [ ] Documentation
 - [x] SVCat CLI flag
 - [ ] Server Flag for config
   - [ ] Chart changes
   - [ ] removing a flag by marking deprecated and hiding to avoid
         breaking the chart release and existing clients who provide a
         flag that will get an error when they try to update

### CODE REVIEW

**Comment by anmolbabu** [comment]
AI on @anmolbabu 
1. Setup Docker locally for build and verification

**Comment by anmolbabu** [comment]
@jberkhahn @mszostok Can you please help resolve https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_service-catalog/2828/pull-service-catalog-test-integration/1317777182298214400#1:build-log.txt%3A2267 not sure how this PR is affecting it

File: cmd/svcat/broker/register_cmd_test.go [hunk 1]
```diff
@@ -111,7 +111,7 @@ var _ = Describe("Register Command", func() {
 			}
 			err := cmd.Validate([]string{"bananabroker", "http://bananabroker.com", "--basic-secret", basicSecret, "--bearer-secret", bearerSecret})
 			Expect(err).To(HaveOccurred())
-			Expect(err.Error()).To(ContainSubstring("cannot use both basic auth and bearer auth"))
```
Comment (jberkhahn) [inline cmd/svcat/broker/register_cmd_test.go:114]: would be nice to have some actual tests for the new stuff added - the new flags, the added case(s) in validate(), and the new secret generation in run() with the pass-off of the new secrets name in opts

**Comment by jberkhahn** [comment]
@jhvhs what are your thoughts on this? I guess being able to pass uname/password in is an ok feature, but kind of concerned about having the broker register command change state other than adding the broker and dependent objects.

File: cmd/svcat/broker/register_cmd_test.go (hunk 1 above)
Comment (anmolbabu) [reply cmd/svcat/broker/register_cmd_test.go:114]: Sure I'll add them

**Comment by jhvhs** [comment]
@jberkhahn I believe it's fine to manage a few additional objects as long as the cli takes full responsibility of the said objects. Namely, if a secret was created by the cli upon registering the broker, it should be destroyed upon de-registering. Also, what is supposed to happen if the secret in question already exists? I believe, that the command must fail with an informative explanation.

**Comment by jhvhs** [comment]
Reached a limit on anonymous dockerhub pulls :-(
/retest

**Pull request closed without merging**